sim/hunter/item_sets_pve.go

```diff
@@ -191,3 +191,170 @@ var ItemSetGiantstalkerPursuit = core.NewItemSet(core.ItemSet{
+					}
+				},
+			}))
+		},
+		// Increases main hand weapon damage by 5%.
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			hunter.OnSpellRegistered(func(spell *core.Spell) {
+				if spell.ProcMask.Matches(core.ProcMaskMeleeMH) {
+					spell.DamageMultiplier *= 1.05
+				}
+			})
+		},
+		// Your periodic damage has a 5% chance to reset the cooldown on one of your Strike abilities. The Strike with the longest remaining cooldown is always chosen.
+		6: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Melee 6P Bonus Trigger",
+				ActionID: core.ActionID{SpellID: 467334},
+				OnPeriodicDamageDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if sim.Proc(0.05, "T2 Melee 6PC Strike Reset") {
+						maxSpell := hunter.RaptorStrike
+
+						for _, strike := range hunter.Strikes {
+							if strike.TimeToReady(sim) > maxSpell.TimeToReady(sim) {
+								maxSpell = strike
+							}
+						}
+
+						maxSpell.CD.Reset()
+						aura.Activate(sim) // used for metrics
+					}
+				},
+			}))
+		},
+	},
+})
+
+var ItemSetDragonstalkerPursuit = core.NewItemSet(core.ItemSet{
+	Name: "Dragonstalker's Pursuit",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Your Aimed Shot deals 20% more damage to targets afflicted by one of your trap effects.
+		2: func(agent core.Agent) {
+			// Implemented in aimed_shot.go
+		},
+		// Your damaging Shot abilities deal 10% increased damage if the previous damaging Shot used was different than the current one.
+		4: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				ActionID: core.ActionID{SpellID: 467312},
+				Label:    "S03 - Item - T2 - Hunter - Ranged 4P Bonus",
+				Duration: time.Second * 12,
+
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.Shots {
+						if hunter.LastShot != nil && spell.SpellCode != hunter.LastShot.SpellCode {
+							spell.DamageMultiplier *= 1.10
+						}
+					}
+				},
+				OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.Shots {
+						if hunter.LastShot != nil && spell.SpellCode != hunter.LastShot.SpellCode {
+							spell.DamageMultiplier /= 1.10
+						}
+					}
+				},
+			})
+
+			core.MakePermanent(hunter.RegisterAura(core.Aura{
+				Label: "S03 - Item - T2 - Hunter - Ranged 4P Bonus Trigger",
+				OnCastComplete: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell) {
+					if spell.Flags.Matches(SpellFlagShot) {
+						procAura.Deactivate(sim)
+						hunter.LastShot = spell
+						procAura.Activate(sim)
+					}
+				},
+			}))
+		},
+		//  Your Serpent Sting damage is increased by 25% of your Attack Power over its normal duration.
+		6: func(agent core.Agent) {
+			agent.(HunterAgent).GetHunter().SerpentStingAPCoeff += 0.25
```

Comment:
Mentioned in DM but I think this needs to be done in an `OnInit` because item code is run before spell code